### ISSUE TITLE
component should setup bindings only "afterPreviousEvents" - P(4) ~8

### ISSUE THREAD

**Issue by justinbmeyer** [canjs/can-component#8]
From: https://github.com/canjs/canjs/issues/1591

It's possible that events are dispatched that represent changes a component's `init` function already knows about.

Putting this another way, in the following code:

```js
init: function(){ 
  // draw out DOM
},
"{scope} change": function(){
  // update DOM
}
```

It's possible that "{scope} change" will be fired after `init`, but it represents a change that `init` already knew about.

With 2.2.3, you can work around this like:

```
init: function(){ 
  can.batch.afterPreviousEvents(() => {
    // draw out DOM
    this.ready = true;
  })
  
},
"{scope} change": function(){
  if(this.ready = true) {
    // update DOM
  }
}
```

Instead, can.Component could only attach bindings "afterPreviousEvents" from the perspective of init.

**Comment by rjgotten** [comment]
@justinbmeyer 
Would be nice if there was a way to bind against a `Map` property or a compute with a templated event binding that would signal `Component` and `Control` to *also* execute the handler at the moment of binding, in addition to the normal behavior of only executing it when a change occurs.

This would very cleanly allow you to set up logic to update the component's DOM that cannot be driven from its view template, without duplicating logic in both `init` as well individual templated event bindings.

For instance, rather than writing:
```js
init : function() {
  this.element.attr( "aria-expanded", this.viewModel.attr( "expanded" ));
}
"{viewModel} expanded" : function( vm, event, value ) {
  this.element.attr( "aria-expanded", value );
}
```

Just write:
```js
"{viewModel} expanded!" : function( vm, event, value ) {
  this.element.attr( "aria-expanded", value );
}
```


This could also avoid a *lot* of overhead that can currently occur when `init` has to retrieve values of computes before they are bound, which means the result of the getter is not cached and has to be re-run when templated event binding are initialized and those computes *do* become bound.

Basically; bind becomes init and thus you can enforce the performance best-practice of bind-before-read.

**Comment by justinbmeyer** [comment]
>  component's DOM that cannot be driven from its view template

Do bindings with `aria-expanded` not work?  Or are you talking about how to setup properties on the component's element from within the component's code?

**Comment by justinbmeyer** [comment]
Not saying it's pretty, but you could write it like:

```js
init : function() {
  this["{viewModel} expanded"](this, {}, this.viewModel.attr( "expanded" ) );
}
"{viewModel} expanded" : function( vm, event, value ) {
  this.element.attr( "aria-expanded", value );
}
```

**Comment by justinbmeyer** [comment]
btw, can you create this in another issue? ... it's related but not the same as this issue.  Thanks!
